local/src/core/http.cc

```diff
@@ -1227,8 +1231,60 @@ Session::set_fd(int fd)
   return errata;
 }
 
+static swoc::Rv<swoc::IPEndpoint>
+get_ip_endpoint_from_device_name(TextView expected_device, int expected_family)
+{
+  swoc::Rv<swoc::IPEndpoint> zret{};
+  auto &ip_endpoint = zret.result();
+
+  struct ifaddrs *ifaddr_list_head = nullptr;
+  if (getifaddrs(&ifaddr_list_head) == -1) {
+    zret.error("getifaddrs failed: {}", swoc::bwf::Errno{});
+    return zret;
+  }
+  bool found_interface = false;
+  for (auto *ifa = ifaddr_list_head; ifa != nullptr; ifa = ifa->ifa_next) {
+    std::string_view interface_name{ifa->ifa_name};
+    if (interface_name != expected_device) {
+      continue;
+    }
+    auto const family = ifa->ifa_addr->sa_family;
+    if (family != expected_family) {
+      continue;
+    }
+    ip_endpoint.assign(ifa->ifa_addr);
+    if (!ip_endpoint.is_valid()) {
+      zret.error("Failed to assign an address from the derived interface.");
+    }
```

Comment:
I suppose this should break here.

Thinking through this further, this functionality would be better encapsulated in a class which can enforce RAII on the allocated struct ifaddrs from getifaddrs.

What do you think of this:
https://github.com/bneradt/proxy-verifier/tree/interface_review2